Python/specialize.c

```diff
@@ -505,6 +514,52 @@ specialize_module_load_attr(
 
 /* Attribute specialization */
 
+void
+_Py_Specialize_LoadSuperAttr(PyObject *global_super, PyObject *class, PyObject *self,
+                             _Py_CODEUNIT *instr, PyObject *name, int load_method) {
+    assert(ENABLE_SPECIALIZATION);
+    assert(_PyOpcode_Caches[LOAD_SUPER_ATTR] == INLINE_CACHE_ENTRIES_LOAD_SUPER_ATTR);
+    _PySuperAttrCache *cache = (_PySuperAttrCache *)(instr + 1);
+    if (!load_method) {
+        SPECIALIZATION_FAIL(LOAD_SUPER_ATTR, SPEC_FAIL_SUPER_NOT_LOAD_METHOD);
+        goto fail;
+    }
+    if (global_super != (PyObject *)&PySuper_Type) {
+        SPECIALIZATION_FAIL(LOAD_SUPER_ATTR, SPEC_FAIL_SUPER_SHADOWED);
+        goto fail;
+    }
+    if (!PyType_Check(class)) {
+        SPECIALIZATION_FAIL(LOAD_SUPER_ATTR, SPEC_FAIL_SUPER_BAD_CLASS);
+        goto fail;
+    }
+    PyTypeObject *tp = (PyTypeObject *)class;
+    PyObject *res = _PySuper_LookupDescr(tp, self, name);
```

Comment:
Is this leaked? I don't see where we DECREF it.

Reply:
Yeah I think it's leaked, _PySuper_LookupDescr incref's its result.

Reply:
For future reference, fix by @JelleZijlstra  in https://github.com/python/cpython/pull/103882